### PR TITLE
:sparkles:feat(llm): add scan WS qr code

### PR DESCRIPTION
**Pull request by LucasWerey**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - WS flow

### 📝 Description

Add the scan of LLD WS Qr code.
Change numbered list to fit with designs

https://github.com/user-attachments/assets/fb9b2378-bd6b-40b7-89db-edfdfea2e995

### ❓ Context

- **JIRA or GitHub link**: [LIVE-12168](https://ledgerhq.atlassian.net/browse/LIVE-12168)

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-12168]: https://ledgerhq.atlassian.net/browse/LIVE-12168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/components/RequiresCameraPermissions/Fallback.tsx [hunk 1]
```diff
@@ -22,11 +23,12 @@ const FallbackCameraBody: React.FC<Props> = ({
   buttonTitle,
   onPress,
   event,
+  hasNoBackground,
 }: Props) => {
   const { colors } = useTheme();
 
   return (
-    <Flex flex={1} bg="background.main" px={6}>
+    <Flex flex={1} bg={hasNoBackground ? null : "background.main"} px={6}>
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/components/RequiresCameraPermissions/Fallback.tsx:31]: "transparent" ?

File: apps/ledger-live-mobile/src/components/RequiresCameraPermissions/index.tsx [hunk 2]
```diff
@@ -16,6 +16,7 @@ type Props = {
    * context, rather than directly showing an error screen.
    * */
   optimisticallyMountChildren?: boolean;
+  fallBackHasNoBackground?: boolean;
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/components/RequiresCameraPermissions/index.tsx:19]:
```suggestion
  fallbackHasNoBackground?: boolean;
```

File: apps/ledger-live-mobile/src/newArch/features/WalletSync/components/Synchronize/ScanTargetSvg.tsx
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/newArch/features/WalletSync/components/Synchronize/ScanTargetSvg.tsx:1]: What is it ?

You can it in package `Icons` ?

File: apps/ledger-live-mobile/src/newArch/features/WalletSync/screens/Activation/ActivationDrawer.tsx [hunk 3]
```diff
@@ -41,11 +40,12 @@ function View({
         hasBackButton={canGoBack}
         onBack={goBackToPreviousStep}
       >
-        <Flex maxHeight={maxDrawerHeight}>
+        <Flex maxHeight={"90%"}>
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/newArch/features/WalletSync/screens/Activation/ActivationDrawer.tsx:43]: Why not 100%, even with 100% you still have a gap between top of screen if I'm correct

File: apps/ledger-live-mobile/src/newArch/features/WalletSync/components/Synchronize/ScanTargetSvg.tsx
Comment (LucasWerey) [reply apps/ledger-live-mobile/src/newArch/features/WalletSync/components/Synchronize/ScanTargetSvg.tsx:1]: That's the svg for the camera border not an Icon

File: apps/ledger-live-mobile/src/newArch/features/WalletSync/screens/Activation/ActivationDrawer.tsx (hunk 3 above)
Comment (LucasWerey) [reply apps/ledger-live-mobile/src/newArch/features/WalletSync/screens/Activation/ActivationDrawer.tsx:43]: Not on some devices (with last iphones and the top bar it doesn't work)
image

**Comment by mcayuelas-ledger** [comment]
Screenshot 2024-08-06 at 16 03 10
it should be like that

**Comment by LucasWerey** [comment]
> Screenshot 2024-08-06 at 16 03 10 it should be like that

@mcayuelas-ledger changed 
image

File: apps/ledger-live-mobile/src/newArch/features/Accounts/screens/AddAccount/components/StepFlow.tsx [hunk 4]
```diff
@@ -49,6 +50,13 @@ const StepFlow = ({
     if (onGoBack) onGoBack(() => setCurrentStep(prevStep => getPreviousStep(prevStep)));
   }, [getPreviousStep, onGoBack]);
 
+  // That means the url as be stored in the store
+  const onQrCodeScanned = (data: string) => {
+    // eslint-disable-next-line no-console
+    console.log(data);
+    // setCurrentStep(Steps.PinCodeInput);
+  };
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/newArch/features/Accounts/screens/AddAccount/components/StepFlow.tsx:58]: Fixed in next PR :)

**Review by mcayuelas-ledger — APPROVED** [review]

**Comment by cgrellard-ledger** [comment]
Screenshot 2024-08-07 at 11 31 07
It looks weird with the corners inside of the black screen instead of how it looks in the figma
Screenshot 2024-08-07 at 11 32 43

**Comment by LucasWerey** [comment]
> It looks weird with the corners inside of the black square instead of how it looks in the figma (at least we could put the same border radius on the black square and on the corners so it doesn't overflow)

Yes I know but in Android the camera can't be rounded so I need to see with Cédric what is the best solution

File: apps/ledger-live-mobile/src/newArch/features/WalletSync/components/Synchronize/QrCode.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      height={"100%"}
+    >
+      <Flex
+        alignItems={"center"}
+        width={QRSize}
+        height={QRSize}
+        maxWidth={280}
+        maxHeight={280}
+        borderRadius={11.52}
+        background={"#fff"}
```
[the remark this answers is not in the document]
Comment (LucasWerey) [reply apps/ledger-live-mobile/src/newArch/features/WalletSync/components/Synchronize/QrCode.tsx:73]: It has been done in previous PR  I need to rebase

File: apps/ledger-live-mobile/src/newArch/features/WalletSync/components/Synchronize/QrCode.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  qrCodeValue: string;
+};
+
+const QrCode = ({ qrCodeValue }: Props) => {
+  const { colors } = useTheme();
+  const { width } = getWindowDimensions();
+  const { t } = useTranslation();
+
+  const QRSize = Math.round(width - 48);
+  const maxQRCodeSize = 280 - 15.36 * 2;
```
[the remark this answers is not in the document]
Comment (LucasWerey) [reply apps/ledger-live-mobile/src/newArch/features/WalletSync/components/Synchronize/QrCode.tsx]: It has been done in previous PR I need to rebase

**Review by cgrellard-ledger — APPROVED** [review]

**Comment by LucasWerey** [comment]
> > It looks weird with the corners inside of the black square instead of how it looks in the figma (at least we could put the same border radius on the black square and on the corners so it doesn't overflow)
> 
> Yes I know but in Android the camera can't be rounded so I need to see with Cédric what is the best solution

@cgrellard-ledger 
image
I think it's better like that also it works for android that way

**Review by cgrellard-ledger — APPROVED** [review]

**Pull request merged**